sensor/common/networkflow/manager/purger.go

```diff
@@ -0,0 +1,284 @@
+	purgerTicker := time.NewTicker(nonZeroPurgerCycle())
+	defer purgerTicker.Stop()
+
+	p := &NetworkFlowPurger{
+		clusterEntities: clusterEntities,
+		manager:         mgr,
+		purgerTicker:    purgerTicker,
+		purgerTickerC:   purgerTicker.C,
+		maxAge:          maxAge,
+		stopper:         concurrency.NewStopper(),
+		purgingDone:     concurrency.NewSignal(),
+	}
+	for _, o := range opts {
+		o(p)
+	}
+	return p
+}
+
+func (p *NetworkFlowPurger) Start() error {
+	var err error
+	if p.manager == nil {
+		err = errors.New("programmer error: network flow purger is not bound to a network flow manager")
+	}
+	// Allow starting the purger without a manager. This is done to prevent blocking of the entire component when
+	// `purgerTickerC` receives a message
+	go p.run()
+	return err
+}
+
+func (p *NetworkFlowPurger) Stop(_ error) {
+	p.purgerTicker.Stop()
+	if !p.stopper.Client().Stopped().IsDone() {
+		defer func() {
+			_ = p.stopper.Client().Stopped().Wait()
+		}()
+	}
+	p.stopper.Client().Stop()
+}
+
+// nonZeroPurgerCycle delivers a non-zero duration to be used in timers (they panic when set with 0 duration)
+func nonZeroPurgerCycle() time.Duration {
+	purgerCycleSetting := env.EnrichmentPurgerTickerCycle.DurationSetting()
+	if purgerCycleSetting > 0 {
+		return purgerCycleSetting
+	}
+	// Disabled purger will wake up every 71 minutes and execute a noop.
+	// We use 71; a prime number higher than 60 (but not too close to it - there maybe many things happening every hour),
+	// so that it is easier to detect and locate a potential source of a problem if something happens every 71 minutes.
+	return 71 * time.Minute
+}
+
+func (p *NetworkFlowPurger) Notify(e common.SensorComponentEvent) {
+	log.Info(common.LogSensorComponentEvent(e, "NetworkFlowPurger"))
+
+	switch e {
+	// Purger could start earlier than this, but we stick to the `SensorComponentEventResourceSyncFinished` as it also
+	// enables the networkFlowManager.
+	case common.SensorComponentEventResourceSyncFinished:
+		d := nonZeroPurgerCycle()
+		p.purgerTicker.Reset(d)
+		log.Debugf("NetworkFlowPurger will execute in %s", d.String())
+	case common.SensorComponentEventOfflineMode:
+		if !features.SensorCapturesIntermediateEvents.Enabled() {
+			p.purgerTicker.Stop()
+		}
+	}
+}
+
+func (p *NetworkFlowPurger) run() {
+	defer p.stopper.Flow().ReportStopped()
+	for {
+		select {
+		case <-p.stopper.Flow().StopRequested():
+			return
+		case _, chanOpen := <-p.purgerTickerC:
+			p.purgingDone.Reset()
+			// Do not execute potentially-expensive purger rules when ticker channel is closed.
+			if chanOpen {
+				p.runPurger()
+			}
+			p.purgingDone.Signal()
+		}
+	}
+}
+
+func (p *NetworkFlowPurger) runPurger() {
+	if env.EnrichmentPurgerTickerCycle.DurationSetting() == 0 {
+		// purger is disabled, but we still want to consume the messages from purgerTickerC
+		return
+	}
```

Comment:
Since the `networkFlowManager` checks `if purger != nil` before calling `Start`, `Stop`, and `Notify` we could simply return a `nil` purger in `NewNetworkFlowPurger`. That way testing the disabled purger can be done with `NewNetworkFlowPurger` and we don't need to check the `purgingDone` signal. As to what happens if the consumer overrides the default ticker with `WithPurgerTicker`, I'd suggest to change the function signature to `func WithPurgerTicker(_ *testing.T, ticker <-chan time.Time) PurgerOption` signaling that this should only be used in tests.

This is not a blocking comment though, I'm ok with keeping it as it is.